### PR TITLE
API-35481-sidekiq-config-526

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

- Adds sidekiq options to 526 workers.
- Sidekiq Pro has an expiry by time, regular Sidekiq does not: [reference](https://github.com/sidekiq/sidekiq/issues/3813)
- Sidekiq will automatically retry a job 25 times over about 20 days, 
   *  ?? so 20/25 = .8 retries/day, so 1.6 or 2 retries would approximately be 2 days of retries ??
   * sidekiq_options retry: 2, retry: true if sidekiq_options expires_in: 48.hours, retry: true is not an option
 

## Related issue(s)

- [API-35481](https://jira.devops.va.gov/browse/API-35481)

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## What areas of the site does it impact?
	modified:   modules/claims_api/app/sidekiq/claims_api/v2/disability_compensation_benefits_documents_uploader.rb
	modified:   modules/claims_api/app/sidekiq/claims_api/v2/disability_compensation_docker_container_upload.rb
	modified:   modules/claims_api/app/sidekiq/claims_api/v2/disability_compensation_pdf_generator.rb

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback
Once deployed our Sidekiq jobs have access to Sidekiq Pro, right? Should we set the Sidekiq options to be the commented out line or the line below?

### CODE REVIEW

File: modules/claims_api/spec/sidekiq/v2/disability_compensation_benefits_documents_uploader_spec.rb [hunk 1]
```diff
@@ -116,4 +116,17 @@
...
     end
   end
+
+  describe 'when an errored job has a time limitation' do
+    it 'logs to the ClaimsApi Logger' do
+      error_msg = 'An error occurred from the BD Uploader Job'
+      msg = { 'args' => [claim.id],
+              'class' => subject,
+              'error_message' => error_msg }
+
```
Comment (rockwellwindsor-va) [inline modules/claims_api/spec/sidekiq/v2/disability_compensation_benefits_documents_uploader_spec.rb:126]: @stiehlrod  Is the logging actually being tested somewhere from these tests?  I see the time limit being tested but there is no expectation set for the log out, which I believe would come from inside the exhausted block in this event?  I also believe the exhausted block is already being tested for these anyways so the logging check is probably not even needed for any of these tests (?)  Let me know if I am missing something there, but the `msg` variable and the `error_msg` can probably just be removed from this since I'm not sure they need to be tested again and it does not look like they are being used in these tests currently.

**Comment by FonzMP** [comment]
this is interesting. the initial goal when we created this ticket was to only enforce a 48 hour retry for the Document Upload to BD, not across the entirety of the flow. but I think this may actually warrant some discussion. I'll kick off a thread in our Dash channel.

File: modules/claims_api/app/sidekiq/claims_api/v2/disability_compensation_benefits_documents_uploader.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@ module ClaimsApi
   module V2
     class DisabilityCompensationBenefitsDocumentsUploader < ClaimsApi::ServiceBase
       LOG_TAG = '526_v2_Benefits_Documents_Uploader_job'
+      sidekiq_options expires_in: 48.hours, retry: true
```
Comment (FonzMP) [inline modules/claims_api/app/sidekiq/claims_api/v2/disability_compensation_benefits_documents_uploader.rb:10]: I'm not fully aware of how `expires_in` behaves but I'd have to imagine that this would not indicate an errored status on the claim upon expiration? I'm not sure if we can set some form of callback or expiration method that would set said status at the end?

TLDR; does this or do we need to find a way to set the claim as `ERRORED`

Comment (FonzMP) [reply modules/claims_api/app/sidekiq/claims_api/v2/disability_compensation_benefits_documents_uploader.rb:10]: Resolution: Good to enforce 2 day retry across all jobs 👍

Comment (rockwellwindsor-va) [reply modules/claims_api/app/sidekiq/claims_api/v2/disability_compensation_benefits_documents_uploader.rb:10]: @FonzMP  I bet we could do that in the [retries exhausted block](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/app/sidekiq/claims_api/service_base.rb#L19) that gets called when this would expire (but also when anything else exhausts its retries which may be ok?)

**Review by FonzMP — APPROVED** [review]
:shipit:
